network/transparent_endpointclient_linux.go

```diff
@@ -109,14 +109,15 @@ func (client *TransparentEndpointClient) AddEndpoints(epInfo *EndpointInfo) erro
 
 	containerIf, err := client.netioshim.GetNetworkInterfaceByName(client.containerVethName)
 	if err != nil {
-		return newErrorTransparentEndpointClient(err.Error())
+		return newErrorTransparentEndpointClient(err)
 	}
 
 	client.containerMac = containerIf.HardwareAddr
+	ep.MacAddress = containerIf.HardwareAddr
```

Comment:
this doesn't look correct to me.. should not be setting ep.MacAddress here.. lets discuss on this

Reply:
moved back to old style where we get containerIf in endpoint_linux.go as discussed offline